service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -348,7 +370,7 @@ private List<RunStateResponse> buildInputsAndSubmitRun(
               .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 
       if (engineIdToWorkflowInput.isEmpty()) {
-        return runStateResponseList;
+        return new ImmutablePair<>(runStateResponseList, List.of());
```

Comment:
Nit: Is this "build the return type" logic is getting complex enough to warrant a separate function and a small set of unit tests?